fedbiomed/researcher/experiment.py

```diff
@@ -1554,39 +1551,224 @@ def run_once(self, increase: bool = False, test_after: bool = False) -> int:
+        return nodes_aux_var
+
+    def _aggregate_cleartext_model_params_and_optim_auxvar(
+        self,
+        model_params: Dict[str, Dict[str, Union[torch.Tensor, np.ndarray]]],
+        optim_auxvar: Optional[Dict[str, Dict[str, AuxVar]]],
+        weights: Dict[str, float],
+    ) -> Tuple[
+        Dict[str, Union[np.ndarray, torch.Tensor]],
+        Optional[Dict[str, AuxVar]],
+    ]:
+        """Aggregate nodes' model parameters and (opt.) optimizer auxiliary variables.
+
+        Args:
+            model_params: Dict of node-wise model parameters.
+            optim_auxvar: Optional dict of node-wise optimizer auxiliary variables.
+            weights: Dict of pre-computed node-wise aggregation weights.
+
+        Returns:
+            aggregated_params: Aggregated model parameters, unflattened into a dict
+                mapping weights' names to their values.
+            aggregated_auxvar: Optional aggregated optimizer auxiliary variables,
+                unflattened into a dict mapping modules' names to `AuxVar` objects
+                they are meant to receive and process.
+        """
+        # Aggregate model parameters.
+        aggregated_params = self._aggregator.aggregate(
+            model_params,
+            weights,
+            global_model=self._global_model,
+            training_plan=self._job.training_plan,
+            training_replies=self._job.training_replies,
+            node_ids=self._job.nodes,
+            n_updates=self._training_args.get('num_updates'),
+            n_round=self._round_current
+        )
+        # Aggregate (opt.) optimizer auxiliary variables.
+        if optim_auxvar:
+            aggregated_auxvar = self._aggregate_cleartext_optim_auxvar(optim_auxvar)
+        else:
+            aggregated_auxvar = None
+        # Return aggregated model parameters and optimizer auxiliary variables.
+        return aggregated_params, aggregated_auxvar
+
+    def _aggregate_cleartext_optim_auxvar(
+        self,
+        nodes_auxvar: Dict[str, Dict[str, AuxVar]],
+    ) -> Dict[str, AuxVar]:
+        """Aggregate clear-text node-wise optimizer auxiliary variables.
+
+        Args:
+            nodes_auxvar: Dict of node-wise optimizer auxiliary variables,
+                with format `{node_name: {module_name: module_auxvar}}`.
+
+        Returns:
+            Aggregated optimizer auxiliary variables, with format
+            `{module_name: module_auxvar}`.
+
+        Raises:
+            FedbiomedExperimentError: If any node sent unproper-type data.
+        """
+        failed = []  # type: List[str]
+        auxvar = {}  # type: Dict[str, AuxVar]
+        for node_name, node_dict in nodes_auxvar.values():
+            if not isinstance(node_dict, dict):
+                failed.append(node_name)
+            elif not failed:
+                for mod_name, mod_auxv in node_dict.items():
+                    auxvar[mod_name] = auxvar.get(mod_name, 0) + mod_auxv
```

Comment:
No, because this is using a trick from the `AuxVar.__add__` method (inherited from its parent `declearn.utils.Aggregate`) that makes it so that a `0` integer is treated as a valid value, and merely means ("return the AuxVar value, discarding the added 0").

Another way to write line 1659 would be:
```python
if mod_name in auxvar:
    aux_var[mod_name] = aux_var[mod_name] + mod_auxv
else:
    aux_var[mod_name] = mod_auxv
```